src/components/ChatApp/Settings/ShareOnSocialMedia.js

```diff
@@ -42,35 +51,10 @@ export default class ShareOnSocialMedia extends Component {
   };
 
   componentDidMount() {
-    this.facebookLoginAPI();
+    this.initFacebookAPI();
   }
 
-  testAPI = () => {
-    window.FB.api('/me', function(response) {});
-    this.setState({ loginStatus: 'connected' });
-  };
-
-  statusChangeCallback = response => {
-    if (response.status === 'connected') {
-      this.testAPI();
-    } else if (response.status === 'not_authorized') {
-      console.log("You're not authorized.");
-    } else {
-      console.log("You're not authorized.");
-    }
-  };
-
-  checkLoginStatus = () => {
-    window.FB.getLoginStatus(response => {
-      this.statusChangeCallback(response);
-    });
-  };
-
-  facebookLogin = () => {
-    window.FB.login(this.checkLoginStatus);
-  };
-
-  Share = () => {
+  shareToFacebook = () => {
     if (
       this.state.loginStatus === 'connected' ||
```

Comment:
The `loginStatus` is not changing, do we require this state?

Reply:
oh, now we can convert this to functional component. Great!